### ISSUE TITLE
From vs. Sender fields - "on behalf of'

### ISSUE THREAD

**Issue by dhanak** [protz/GMail-Conversation-View#954]
It would be great if the extension could support and distinguish between "from" and "sender" fields. Currently, the "sender" field is not displayed at all. But if it is present, it should be displayed as follows:

`John Doe <john.doe@domain.com> on behalf of Jane Deer <jane.deer@domain.com> to Me <myself@company.org>`

where John Doe is the "sender" and Jane Deer is the "from". Judging from the add-on source, this seems to be a minor modification, as the information is already there - If I am not mistaken, a new contact object needs to be created from the "realFrom" address and passed to the template.

Thanks in advance!

**Comment by protz** [comment]
Yeah that's what Outlook does. What does Thunderbird do by default on 
such emails?

**Comment by dhanak** [comment]
Not much I am afraid; this issue has been there for more than 10 years! See https://bugzilla.mozilla.org/show_bug.cgi?id=221054 for details. The related issue https://bugzilla.mozilla.org/show_bug.cgi?id=285474 mentions a hidden option called `mailnews.headers.showSender`. If I turn this on, the Sender field is displayed in the header separately, that's all.

**Comment by protz** [comment]
I find the "on behalf of" feature of Outlook to take a lot of screen 
space for (imho) rarely useful information. Also, both bugs you point to 
have been fixed, and there doesn't seem to be any consensus in 
Thunderbird that we want to do this "on behalf of" thing.

So, for the sake of consistency, I'd much rather stick to what 
Thunderbird does, unless you point me to a (recent) bug or "official" 
discussion where we say "we want this".

Thanks,

Jonathan

**Comment by dhanak** [comment]
For me, at least, this information would be very useful. These sender fields tend to crop up in forwarded Outlook calendar invites, where I only see the meeting organizer in the from field, whereas the showing forwarder would also be informative; especially if she adds extra notes to the invite, in which case the message becomes confusing without it. Still, I respect and understand your point of view.

One of the bugs have been resolved as duplicate, the other was about something else, and I personally find the resolution less than satisfactory (seriously, a hidden preference disabled by default?). Besides, both discussions are ancient, and I couldn't find a more recent discussion on the issue.

Could you perhaps implement the "on behalf of" feature guarded with an option? Or at least display the sender field in the detailed header view, as TB does with the pref turned on? Then I would still have the option to tweak the template in my own profile to display the relevant information as I see fit.

Thanks!

**Comment by MichaelInParis03** [comment]
Hello,

Regardless of the value of mailnews.headers.showSender, TB displays the value of the Sender: field (if present) of the mail header in the From column of the message list, as well as in the detailed view of the message, always overriding the value of the From: filed of the mail header.

This is a problem for a mailing list that I run, which puts its address in the Sender: field. As a consequence, looking at the list of messages sent to the list, they all show as originating from the list rather than from the subscriber who sent it, whose email address and name are in the From: field.

Is there a way to fix this, and have TB display the From: field value?

I am using the French version of TB version 52.6.0,
